### PR TITLE
chore(build): migrate build from travis to github actions

### PR DESCRIPTION
**Pull request by akhilerm**
**Why is this PR required? What issue does it fix?**:
Migrates the cstor build and test process from travis to github actions

**What this PR does?**:
- add scripts for installing dependencies (tools and build)
- add build and release workflow
- move downstream release tagging to github actions

**Does this PR require any upgrade changes?**:
No

**If the changes in this PR are manually verified, list down the scenarios covered and commands you used for testing with logs:**

**Any additional information for your reviewer?**:
_Mention if this PR is part of any design or a continuation of previous PRs_
`GR_TOKEN` secret need to be added to the repository for downstream tagging.

**Checklist:**
- [ ] Fixes #<issue number>
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [x] Has the change log section been updated?
- [ ] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track:
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

### CODE REVIEW

**Review by mynktl — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Build & CI
+
+on:
+  push:
```
Comment (shubham14bajpai) [inline .github/workflows/build.yml:4]: Should we have on create as well? If you create a new branch then this workflow will not get executed.

Comment (akhilerm) [reply .github/workflows/build.yml:4]: @shubham14bajpai If you check this, you can see that for new branch creation , the workflow is triggered. https://github.com/akhilerm/cstor/actions

Comment (akhilerm) [reply .github/workflows/build.yml:4]: But I think for release, we need to create, as its not being triggered.

**Comment by akhilerm** [comment]
@mynktl , Should we have a release workflow for this repo? Since there is no image generation, checks that will be performed in release will already be done as part of build when a commit is pushed.. cc: @shubham14bajpai

**Comment by mynktl** [comment]
> @mynktl , Should we have a release workflow for this repo? Since there is no image generation, checks that will be performed in release will already be done as part of build when a commit is pushed.. cc: @shubham14bajpai

@akhilerm yes, we can disable the workflow on release tagging. If it saves the release creation time, then better to do it.

**Review by shubham14bajpai — APPROVED** [review]
LGTM

**Comment by akhilerm** [comment]
@mynktl can this be merged now?

**Review by mynktl — APPROVED** [review]

**Pull request merged**